warehouse/warehouse.go

```diff
@@ -1195,11 +1195,10 @@ func pendingEventsHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	sourceID := pendingEventsReq.SourceID
-
+	sourceID, taskRunID := pendingEventsReq.SourceID, pendingEventsReq.TaskRunID
 	// return error if source id is empty
-	if sourceID == "" {
-		pkgLogger.Errorf("[WH]: pending-events:  Empty source id")
+	if sourceID == "" || taskRunID == "" {
+		pkgLogger.Errorf("[WH]: pending-events:  Empty source_id or task_run_id ")
```

Comment:
Can we make it a structured log? Also, shouldn't it be a warning?

Reply:
Why a warning and not error ?

Reply:
Since the error is not causing our application to be in trouble or involving us in the investigation immediately. I thought it was okay to mark them as a warning.
Also, since these APIs are being polled continuously by the extract, we might see too much noise as well.